bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -1491,6 +1492,40 @@
 		_t.input.addEventListener("change", onChange);
 	}
 
+	/* class ControlInput extends Control */
+	function ControlInput(parentNode) {
+		Control.call(this, parentNode);
+
+		var
+			_t = this;
+
+		_t.input = _t.parentNode.querySelector("input[type=text]");
+
+		_t.valueNode = _t.parentNode.parentNode.querySelector(o.formValue);
+
+		function onChange() {
+			_t.parentNode.dispatchEvent(createEvent("control-change", {
+				item: _t.item,
+				value: _t.input.value
+			}));
+		}
+
+		_t.setValuePrivate = function(value) {
+			_t.input.value = value;
```

Comment:
In Java, we are considering value nad item state. Not there ?
If necessary, you can consider this:
```
		_t.setValuePrivate = function(value, itemState) {
		}
```

Reply:
The one thing that one cannot do at the moment is make the field completely empty if the underlying item type is numeric or quantityType. One would need to be able to post UNDEF, but I don't think that is useful, and it would block the possibility to update Text items with the string UNDEF. So I don't think there is real need to use itemState here.

Reply:
At least, in my testing, it behaved as I expected without.

Reply:
I removed item state from java, only kept a check for null or UNDEF.

Reply:
Isee that it is similar for the Text widget, so I think it is OK.